### PR TITLE
Switch Documenter.jl math engine to MathJax3 + Doc Fixes

### LINKED ISSUE #528: Equation numbering in docs broken
Screen Shot 2019-11-08 at 7 21 15 AM

### PR DESCRIPTION
**Pull request by ali-ramadhan**
Documenter.jl uses KaTeX by default but MathJax3 might be better for us. Will wait for Buildkite to build and deploy the docs to see 🤞 

Resolves #528 (hopefully)

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -141,7 +141,8 @@ pages = [
 format = Documenter.HTML(
     collapselevel = 1,
        prettyurls = get(ENV, "CI", nothing) == "true",
-        canonical = "https://clima.github.io/OceananigansDocumentation/stable/"
+        canonical = "https://clima.github.io/OceananigansDocumentation/stable/",
+       mathengine = MathJax3()
```
Comment (glwagner) [inline docs/make.jl:145]: It's that easy!

**Review by glwagner — APPROVED** [review]

**Comment by navidcy** [comment]
Did you view the preview docs?

**Comment by glwagner** [comment]
> Did you view the preview docs?

Ah, doesn't look great...

**Review by glwagner — CHANGES_REQUESTED** [review]
Ok, this probably can't be merged because it looks like this

image

File: docs/make.jl (hunk 1 above)
Comment (navidcy) [reply docs/make.jl:145]: I doubted it would be

**Comment by ali-ramadhan** [comment]
Ah it was just a try. `\bm` comes from the `bm` LaTeX package which I guess was available with KaTeX. For MathJax we might have to define the `\bm` command at the top of the pages that use it?

[Equation numbering still doesn't work](https://clima.github.io/OceananigansDocumentation/previews/PR1530/numerical_implementation/time_stepping/) but apparently it's because MathJax3 doesn't enable it by default: http://docs.mathjax.org/en/latest/input/tex/eqnumbers.html

Looks like we can configure it by passing a dict to the `MathJax3` constructor though: https://juliadocs.github.io/Documenter.jl/stable/lib/internals/writers/#Documenter.Writers.HTMLWriter.MathJax3

**Comment by navidcy** [comment]
I started fixing it. Seems like `\underbrace{}` doesn't compile in MathJax3... A quick googling didn't reveal any solutions. Will try to look into it later and continue. :)

**Comment by navidcy** [comment]
@ali-ramadhan, I believe this is working now. There are some issues when referencing other sections of the docs, but perhaps let's leave that for a future PR? Equation numbering **within each `.md` file** works now!

**Comment by navidcy** [comment]
hm... I merged master and the unit tests seem to fail saying

```julia

[2021/04/09 19:34:17.876] WARN  Checksum did not match -@-> /storage7/buildkite-agent/.julia-1983/packages/DataDeps/ooWXe/src/verification.jl:24
--
  | Do you wish to Abort, Retry download or Ignore
  | [a/r/i]
```

Apparently buildkite is not answering so tests stop at that point indefinitely...

https://buildkite.com/clima/oceananigans/builds/1983#a259cdda-aa8b-40f1-861f-e198c1e7c7f2

**Comment by ali-ramadhan** [comment]
@navidcy Ah sorry about that error, it was fixed in PR #1552. I merged `master` into this branch which should fix the unit tests.

**Comment by navidcy** [comment]
hm... tests pass here after 3-4 days!
weird.
seems like just something made the code extremely slow?

**Comment by glwagner** [comment]
None of the individual test runs took very long, but our CI pipeline is clogged up due to this hanging issue.

**Comment by ali-ramadhan** [comment]
Looks like this PR upgrades to Julia 1.6 as well now?

**Comment by navidcy** [comment]
Yes.... it was part of my attempts to de-clog everything.
But I'm thinking to revert those changes!

File: .gitignore [hunk 2]
```diff
@@ -10,7 +10,6 @@ deps/deps.jl
 *~
 
 # Documenter.jl and MkDocs
-docs/Manifest.toml
```
Comment (navidcy) [inline .gitignore:13]: @ali-ramadhan should `docs/Manifest.toml` be in the `.gitignore` or was I right to take it ou?

**Comment by navidcy** [comment]
OK. We have a preview!

https://clima.github.io/OceananigansDocumentation/previews/PR1530/

@ali-ramadhan, could you have a look?

**Review by navidcy — APPROVED** [review]
I'm approving here but I made so many changes that, in principle, I should be asking for others to review this :)

**Comment by navidcy** [comment]
We have a preview at https://clima.github.io/OceananigansDocumentation/previews/PR1530/ which revealed some latexing issues... I'll get to them :)

**Comment by navidcy** [comment]
See the preview @ https://clima.github.io/OceananigansDocumentation/previews/PR1530/

**Comment by ali-ramadhan** [comment]
Thanks so much @navidcy for putting all this work into making the docs beautiful again!

I guess I can't approve this PR since I opened it but here's a thumbs up 👍 PR needs @glwagner's approval since he requested changes I think.

**Comment by navidcy** [comment]
I don't think @glwagner requested anything.

There are still some links in the docs that are broker (e.g., couple refs to eqs that don't exist). With MathJax3 `\label{..}` and `\eqref{..}` works *but only within the same page*. I tried to minimise referencing equations from other pages but there may be still some left. I'll fix them in a different PR though, together with some links to sections that may not exist anymore. We should do any refs to other docs sections via Documenter's functionality so then it'll complain when something breaks.

File: docs/src/appendix/fractional_step.md [hunk 3]
```diff
@@ -1,44 +1,45 @@
 # Fractional step method
 
-Solving the momentum equation \eqref{eq:momentumFV} coupled with the continuity equation \eqref{eq:continuityFV} can be
-cumbersome so instead we employ a fractional step method. To approximate the solution of the coupled system we first
-solve an approximation to the discretized momentum equation \eqref{eq:momentumFV} for an intermediate velocity field
-``\bm{u}^\star`` without worrying about satisfying the incompressibility constraint. We then project ``\bm{u}^\star`` onto
-the space of divergence-free velocity fields to obtain a value for ``\bm{u}^{n+1}`` that satisfies
-\eqref{eq:continuityFV}.
+Solving the momentum equation \eqref{eq:momentumFV} coupled with the continuity equation can
```
Comment (francispoulin) [inline docs/src/appendix/fractional_step.md:3]: I see this is where the referencing to equations does not work.  I am fine with having this fixed in another PR if that's easier.

**Comment by ali-ramadhan** [comment]
That makes sense. Will make sure to use Documenter.jl's referencing for referencing across pages.

> I don't think @glwagner requested anything.

Ah I thought merging is blocked right now because someone requested changes in their PR review. Maybe I'm seeing something different as the one who opened the PR though.

image

**Comment by navidcy** [comment]
oh yes...

@glwagner we hereby request your re-review and, hopefully, your approval.

**Review by francispoulin — APPROVED** [review]
An impressive number of changes @navidcy, very well done!  Everything looks good to me except the referencing the equations that you mentinoed.  Also, @ali-ramadhan had a comment and not sure if that discussion was resolved or not.

**Comment by navidcy** [comment]
thanks @francispoulin
regarding @ali-ramadhan's remark: yeap, I've reverted the Julia v1.6 transition.

**Comment by ali-ramadhan** [comment]
Woah did not know you could do this :eyes:

image

**Comment by navidcy** [comment]
> Woah did not know you could do this 👀
> 
> image

I didn't know either! But I saw the button and thought it was appropriate at this point so I'd give it a go.
@glwagner didn't even specify what changes needed to be made! haha
